### PR TITLE
Add article about video time estimation

### PR DESCRIPTION
**Pull request by kjcpaas**
Hi @quipper/devs I wrote an article about the problem encountered [here](https://github.com/quipper/quipper/issues/8365#issuecomment-276871199)

Enjoy!

PS: I set the date to `March 3, 2017` so it will show after Aya release of new Qlink (as some of the images shown are for new QLink). Hence it will not show if you try in locally. If you want it to show, please adjust the date to an earlier date (e.g. `2017-02-26`)

### CODE REVIEW

File: _posts/2017-03-03-estimating-video-time-by-linear-regression.markdown [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+
+As shown in the table, there are still some cases in which the estimated time is not the same as the time shown by the player. Since this is an estimate, it's impossible to be exact. However, we managed to achieve our goal of getting within 1 second of the actual player video time. :tada:
+
+## Conclusion
+
+From solving this problem, we can learn the following things:
+
+- Video conversion does not only affect the video quality or size, but also the time.
+- Part of a good UI is consistency in the displayed information. No matter how good-looking the UI is, if the information displayed is inconsistent, it ca cause a lot of confusion.
```
Comment (kristm) [inline _posts/2017-03-03-estimating-video-time-by-linear-regression.markdown]: typo: it `can` cause a lot...

Comment (kjcpaas) [inline _posts/2017-03-03-estimating-video-time-by-linear-regression.markdown]: Fixed: https://github.com/quipper/quipper.github.io/pull/19/commits/e88d9197b03674a21370032727fafea4d5529230

**Comment by kjcpaas** [comment]
ping @quipper/devs!

File: _posts/2017-03-03-estimating-video-time-by-linear-regression.markdown [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+comments: true
+---
+
+In Quipper, we use [Brightcove](https://www.brightcove.com/) to host our video contents.
+
+![Quipper Video Sample](https://cloud.githubusercontent.com/assets/3772828/23352748/eb3774a2-fd03-11e6-9fbc-5b4049d986d5.png)
+
+In most cases, the only time we need to show the video time is when the user watches the video. For this, showing the video time in Brightcove's video player itself is enough.
+
+We have a recent update in which we have to show the video time outside of the video player. In this case, we have to get the video time from Brightcove's API. This is when we discovered that the time shown in Brightcove player, and the time from Brightcove API does not match.
```
Comment (bdesmero) [inline _posts/2017-03-03-estimating-video-time-by-linear-regression.markdown]: Not 100% sure but I think the comma is not needed here since we're comparing the 2 🤔 
```
This is when we discovered that the time shown in the Brightcove player does not match the one from the Brightcove API.
```
WDYT? 🤔

File: _posts/2017-03-03-estimating-video-time-by-linear-regression.markdown [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+As shown in the table, there are still some cases in which the estimated time is not the same as the time shown by the player. Since this is an estimate, it's impossible to be exact. However, we managed to achieve our goal of getting within 1 second of the actual player video time. :tada:
+
+## Conclusion
+
+From solving this problem, we can learn the following things:
+
+- Video conversion does not only affect the video quality or size, but also the time.
+- Part of a good UI is consistency in the displayed information. No matter how good-looking the UI is, if the information displayed is inconsistent, it can cause a lot of confusion.
+- Even though web developers don't usually use Math in their line of work, they should be prepared to apply the concept to solve the problems encountered.
```
Comment (rbmrclo) [inline _posts/2017-03-03-estimating-video-time-by-linear-regression.markdown]: This sounds very opinionated, are you sure about this? :thinking:

File: _posts/2017-03-03-estimating-video-time-by-linear-regression.markdown [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+![Quipper Video Sample](https://cloud.githubusercontent.com/assets/3772828/23352748/eb3774a2-fd03-11e6-9fbc-5b4049d986d5.png)
+
+In most cases, the only time we need to show the video time is when the user watches the video. For this, showing the video time in Brightcove's video player itself is enough.
+
+We have a recent update in which we have to show the video time outside of the video player. In this case, we have to get the video time from Brightcove's API. This is when we discovered that the time shown in Brightcove player, and the time from Brightcove API does not match.
+
+![Time mismatch](https://cloud.githubusercontent.com/assets/3772828/22539207/61f06e4c-e953-11e6-98ba-7756d09f2a93.png)
+
+In the image above, the video player shows `18:35`, while the navigation button shows `18:32` (this comes from the API). The time in video player is off by **3 seconds**! This can confuse the users so we have to do something for the video times to match, or at least be off by just 1 second.
+
```
Comment (eugiellimpin) [inline _posts/2017-03-03-estimating-video-time-by-linear-regression.markdown]: How about `video timestamps`? 🤔

File: _posts/2017-03-03-estimating-video-time-by-linear-regression.markdown [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+This relationship can be visualized by a graph:
+
+![Graph](https://cloud.githubusercontent.com/assets/3772828/23353842/7946a3da-fd09-11e6-98d4-d2c2c46cd877.png)
+
+The actual relationship might not be linear at all but for the usual time range of our videos (4 minutes to 40 minutes), assuming a linear relationship is enough.
+
+## Solution
+
+The approach was to find the line that would best fit the relationship between the source video time and the converted video time. We got some sample videos, plotted their times in a table.
```
Comment (bdesmero) [inline _posts/2017-03-03-estimating-video-time-by-linear-regression.markdown]:
```
We got some sample videos and plotted their times in a table.
```
WDYT?

File: _posts/2017-03-03-estimating-video-time-by-linear-regression.markdown [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+From this, we got the best fit line with the equation `y = 1.002x + 0.4253`, where `y` is **converted video time**, and `x` is **source video time**.
+
+![Best fit line](https://cloud.githubusercontent.com/assets/3772828/22541505/a2a02af0-e961-11e6-9a7e-1cdfb17f37af.png)
+
+We adjusted the `y-intercept` to `0.7` from `0.4253` to take into account that our API applies `floor` function to store the the time in seconds (hence the actual source time is longer by less than 1 second).
```
Comment (bdesmero) [inline _posts/2017-03-03-estimating-video-time-by-linear-regression.markdown]: I think the thought flow will be easier when you start with the beginning value 🤔 There's an extra `the` here as well.
```
We adjusted the `y-intercept` from `0.4253` to `0.7` to take into account the `floor` function our API applies to store the time in seconds ...
```

File: _posts/2017-03-03-estimating-video-time-by-linear-regression.markdown [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+![Table](https://cloud.githubusercontent.com/assets/3772828/22541488/7b634d0a-e961-11e6-91d0-109801c0ce38.png)
+
+
+As shown in the table, there are still some cases in which the estimated time is not the same as the time shown by the player. Since this is an estimate, it's impossible to be exact. However, we managed to achieve our goal of getting within 1 second of the actual player video time. :tada:
+
+## Conclusion
+
+From solving this problem, we can learn the following things:
+
+- Video conversion does not only affect the video quality or size, but also the time.
```
Comment (bdesmero) [inline _posts/2017-03-03-estimating-video-time-by-linear-regression.markdown]: Video `time` refers to the total `length` of the video, right?

File: _posts/2017-03-03-estimating-video-time-by-linear-regression.markdown (hunk 3 above)
Comment (kjcpaas) [inline _posts/2017-03-03-estimating-video-time-by-linear-regression.markdown]: Well, I just realized that we are not using enough Math to my personal liking (:sweat_smile) but we are still using some. I'll rephrase this to something more general, e.g. applying concepts from other fields.

File: _posts/2017-03-03-estimating-video-time-by-linear-regression.markdown (hunk 4 above)
Comment (kjcpaas) [inline _posts/2017-03-03-estimating-video-time-by-linear-regression.markdown]: I don't think `timestamp` is the right term for this. How about `video length` instead?

Comment (kristm) [inline _posts/2017-03-03-estimating-video-time-by-linear-regression.markdown]: duration?

Comment (kjcpaas) [inline _posts/2017-03-03-estimating-video-time-by-linear-regression.markdown]: duration is much better. 😸

**Comment by kjcpaas** [comment]
@kristm @eugiellimpin @bdesmero @rbmrclo Updated according to your suggestions. Please check again 😸

**Comment by bdesmero** [comment]
`Video duration` sounds better!

File: _posts/2017-03-03-estimating-video-duration-by-linear-regression.markdown [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+
+This is because we are actually dealing with 2 video files instead of 1. These videos are the following:
+
+- Source video
+
+This is the one we uploaded to Brightcove. Its video duration is the one that can be obtained from Brightcove API.
+
+- Converted Video
+
+Once the video is uploaded, it gets converted by Brightcove to be compatible with their player. Due to the conversion process, the video duration cannot be exactly the same as the source videos. The time of this converted video is the one shown in the video player.
```
Comment (bdesmero) [inline _posts/2017-03-03-estimating-video-duration-by-linear-regression.markdown]: `source video`

File: _posts/2017-03-03-estimating-video-duration-by-linear-regression.markdown [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+
+## Observations
+
+We observed other videos and found some patterns:
+
+1. Converted video duration is **longer** than source video duration.i
```
Comment (bdesmero) [inline _posts/2017-03-03-estimating-video-duration-by-linear-regression.markdown]: There's an extra `i` at the end of this sentence 😁

File: _posts/2017-03-03-estimating-video-duration-by-linear-regression.markdown (hunk 8 above)
Comment (kjcpaas) [inline _posts/2017-03-03-estimating-video-duration-by-linear-regression.markdown]: This should be `course video's` 😞

File: _posts/2017-03-03-estimating-video-duration-by-linear-regression.markdown (hunk 9 above)
Comment (kjcpaas) [inline _posts/2017-03-03-estimating-video-duration-by-linear-regression.markdown]: Vim issues 😅

**Comment by kjcpaas** [comment]
@bdesmero Fixes. 😸

File: _posts/2017-03-03-estimating-video-duration-by-linear-regression.markdown [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+
+This relationship can be visualized by a graph:
+
+![Graph](https://cloud.githubusercontent.com/assets/3772828/23353842/7946a3da-fd09-11e6-98d4-d2c2c46cd877.png)
+
+The actual relationship might not be linear at all but for the usual duration range of our videos (4 minutes to 40 minutes), assuming a linear relationship is enough.
+
+## Solution
+
+The approach was to find the line that would best fit the relationship between the source video duration and the converted video duration. We got some sample videos are plotted their duration in a table.
```
Comment (bdesmero) [inline _posts/2017-03-03-estimating-video-duration-by-linear-regression.markdown]: `We got some sample videos and plotted their durations in a table.`?

Comment (kjcpaas) [inline _posts/2017-03-03-estimating-video-duration-by-linear-regression.markdown]: Fixed! https://github.com/quipper/quipper.github.io/pull/19/commits/2e4701aa51122be5f83f411de4d133a32be89361

Comment (bdesmero) [inline _posts/2017-03-03-estimating-video-duration-by-linear-regression.markdown]: Thanks! 🎉

**Comment by bdesmero** [comment]
Thanks @kjcpaas 💃

**Comment by kjcpaas** [comment]
Thanks for taking the time to review @kristm @bdesmero @rbmrclo @eugiellimpin!
